plugins/dell-dock/fu-dell-dock-plugin.c

```diff
@@ -38,72 +46,93 @@ fu_dell_dock_plugin_create_node(FuPlugin *plugin, FuDevice *device, GError **err
 	return TRUE;
 }
 
+static guint8
+fu_dell_dock_plugin_get_dock_type(FuDevice *device, GError **error)
+{
+	if (FU_IS_DELL_DOCK_EC(device))
+		return fu_dell_dock_get_dock_type(device);
+
+	return DOCK_BASE_TYPE_UNKNOWN;
+}
+
 static gboolean
-fu_dell_dock_plugin_probe(FuPlugin *plugin, FuDevice *proxy, GError **error)
+fu_dell_dock_plugin_probe_ec_v1_subcomponents(FuPlugin *plugin, FuDevice *ec_device, GError **error)
 {
-	const gchar *instance_id_mst;
-	const gchar *instance_id_status;
-	g_autofree const gchar *instance_guid_mst = NULL;
-	g_autofree const gchar *instance_guid_status = NULL;
-	g_autoptr(FuDellDockEc) ec_device = NULL;
-	g_autoptr(FuDellDockMst) mst_device = NULL;
-	g_autoptr(FuDellDockStatus) status_device = NULL;
 	FuContext *ctx = fu_plugin_get_context(plugin);
 
-	/* create ec endpoint */
-	ec_device = fu_dell_dock_ec_new(proxy);
-	if (!fu_dell_dock_plugin_create_node(plugin, FU_DEVICE(ec_device), error))
-		return FALSE;
+	/* MST */
+	{
```

Comment:
what's up with this indentation blocks for all these?  Seems totally unnecessary

Reply:
it's for readibility and maintainability for multiple devices in one place.